### DISCUSSION TITLE
FlexSearch configuration to honor canonifyURL?

### DISCUSSION THREAD

**Discussion by Aman-Aalam** [gethinode/hinode#829] [Help] [answered]
Hi Team,

Huge thanks for all the help, because of which I am now about to deploy my site live.

But I am using `canonifyURLS = true` because I am deploying under a subfolder.
Everything works well except the search, where the post URLs are still relative and hence they resolve with the base URL and no canonified URL.

I see Hinode uses Flex Search, is there a setting where I can configure this, or fix this?

**Comment by markdumay** [comment, marked as answer]
I have released [mod-flexsearch v1.10.0](https://github.com/gethinode/mod-flexsearch/releases/tag/v1.10.0) that includes a new [configuration option](https://github.com/gethinode/mod-flexsearch#configuration). Upgrade your module reference to `mod-flexsearch` to this version and add the following configuration to your site configuration (e.g. `hugo.toml`):

```toml
[params.modules.flexsearch]
    canonifyURLs = true
```

In this mode, the search index will use absolute URLs instead of relative URLs. Hopefully this will resolve your issue.

**Reply by Aman-Aalam** [reply]
Great, thanks!
I manually added this new module and the search is working fine now.

One last question, how do I exclude a page from search?
With the new version, all files in the content folder are included while in the older one (`1.8.1`) contents only in the `posts` folder seemed to be indexed.

Will all this end up in the documentation (if not already) ?

**Comment by markdumay** [comment]
You can [exclude individual pages from the search index](https://gethinode.com/docs/content/content-management/#front-matter) by adding the following setting to the page's frontmatter:

```yaml
---
searchExclude: true
---
```

The flex search options are documented in the [mod-flexsearch](https://github.com/gethinode/mod-flexsearch) repo. They are not included in the main docs yet...

**Reply by Aman-Aalam** [reply]
Perfect.
That worked!

Would you mind if I did a PR on the docs repo for Search related documentation?

**Reply by markdumay** [reply]
> Would you mind if I did a PR on the docs repo for Search related documentation?

That would be great!
